### ISSUE TITLE
Write transitions to the `transition` parameter of the `-info` doc in all cases the transition is run

### ISSUE THREAD

**Issue by SCdF** [medic/cht-core#6431]
**What feature do you want to improve?**
We have a log of transitions that have run on documents, stored at `infoDoc.transitions`.

**Describe the improvement you'd like**
Currently it only adds a transition to the log if that transitions `onMatch` function returns `true`. Some don't however, because this boolean does double duty: it also indicates that the transition edited the document and so the document should be saved back into CouchDB at the end of the transition run.

We should split these two concepts so that we can independently declare that a transition ran (and so should be logged) and that it changed the doc (and so should have an entry added). We could do this by returning two things, or more likely just by presuming if `onMatch` ran at all then that counts as a run.

**Comment by dianabarsan** [comment]
> or more likely just by presuming if onMatch ran at all then that counts as a run.
If the `onMatch` does double duty and also works as a `filter` (when filtering requires async operations), maybe we should just go with the 1st idea of returning 2 things.

**Comment by SCdF** [comment]
Yeah that is a good point. We have mused about dropping filter altogether, since it's not very flexible. Could do both here.
